src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -992,9 +985,34 @@ ReactDOMComponent.Mixin = {
         DOMPropertyOperations.deleteValueForProperty(getNode(this), propKey);
       }
     }
+
+    var ordered = DOMProperty.order[this._tag];
+    if (ordered != null) {
+      for (var i = 0, len = ordered.length; i < len; i++) {
+        propKey = ordered[i];
+
+        if (nextProps.hasOwnProperty(propKey)) {
+          lastProp = lastProps ? lastProps[propKey] : undefined;
+          nextProp = nextProps[propKey];
+
+          if (lastProp !== nextProp) {
+            DOMPropertyOperations.setValueForProperty(
+              getNode(this),
+              propKey,
+              nextProp
+            );
+          }
+        }
+      }
+    }
+
     for (propKey in nextProps) {
-      var nextProp = nextProps[propKey];
-      var lastProp =
+      if (ordered != null && ordered.indexOf(propKey) >= 0) {
```

Comment:
I'll run a benchmark. In an earlier version of this PR I stored the properties in an array _and_ an object to eliminate indexOf calls, but I removed it because it added a extra parts and primitive benchmarks of `indexOf` weren't compelling (especially with the general performance improvement by eliminating work in ReactDOMInput).

So let's find out!